raft_test.go

```diff
@@ -1366,11 +1366,11 @@ func TestHandleHeartbeat(t *testing.T) {
 // TestHandleHeartbeatResp ensures that we re-send log entries when we get a heartbeat response.
 func TestHandleHeartbeatResp(t *testing.T) {
 	storage := newTestMemoryStorage(withPeers(1, 2))
+	storage.SetHardState(pb.HardState{Term: 3, Vote: 1, Commit: 3})
 	storage.Append([]pb.Entry{{Index: 1, Term: 1}, {Index: 2, Term: 2}, {Index: 3, Term: 3}})
```

Comment:
are these entries necessary for the test? Maybe both `storage.SetHardState` and `storage.Append` can be removed to keep things simpler